### PR TITLE
(feat) adding setup command

### PR DESCRIPTION
**Pull request by tphoney**
image
image

``` 

curl -d '{"id":"id1", "correlation_id":"this-is-unique", "pool_id":"ubuntu", "setup_request": {"network": {"id":"drone"}, "platform": { "os":"ubuntu" }}}' -H "Content-Type: application/json" -X POST  http://127.0.0.1:3000/setup
{"instance_id":"i-06414bc7c8cfb67b4","ip_address":"18.218.59.14"}
➜  drone-plugin-index git:(fossa) curl -d '{"id":"unique-stage-id","instance_id":"i-06414bc7c8cfb67b4","pool_id":"ubuntu","correlation_id":"i-08e1c992260957c45", "start_step_request":{"id":"step4", "image": "alpine:3.11", "working_dir":"/tmp", "run":{"commands":["sleep 30"], "entrypoint":["sh", "-c"]}}}' -H "Content-Type: application/json" -X POST  http://127.0.0.1:3000/step

{"error_msg":"cannot get the instance by stageId: manager: instance not found"}
➜  drone-plugin-index git:(fossa) curl -d '{"id":"id1","instance_id":"i-06414bc7c8cfb67b4","pool_id":"ubuntu","correlation_id":"i-08e1c992260957c45", "start_step_request":{"id":"step4", "image": "alpine:3.11", "working_dir":"/tmp", "run":{"commands":["sleep 30"], "entrypoint":["sh", "-c"]}}}' -H "Content-Type: application/json" -X POST  http://127.0.0.1:3000/step

{"exited":true}
➜  drone-plugin-index git:(fossa) curl -d '{"id":"id1","pool_id":"ubuntu","correlation_id":"uvw3", "instance_id":"i-06414bc7c8cfb67b4"}' -H "Content-Type: application/json" -X POST  http://127.0.0.1:3000/destroy
➜  drone-plugin-index git:(fossa) date
Wed May  4 12:05:19 BST 2022
```

### CODE REVIEW

**Comment by eoinmcafee00** [comment]
It all looks good to me, only request anywhere you have "LE" it's confusing for anyone reading the code, they've no idea what LE stands for. We should use the full name or drop it so something simple 'helper.getClient'. There's no reason anyone needs to know about the lite engine in this repo.

File: command/delegate/delegate.go [hunk 1]
```diff
@@ -477,7 +476,7 @@ func (c *delegateCommand) handleStep(w http.ResponseWriter, r *http.Request) {
 	logr = logr.
 		WithField("ip", inst.Address)
 
-	client, err := c.getLEClient(inst)
+	client, err := lehelper.GetLEClient(inst, c.runnerName)
```
Comment (eoinmcafee00) [inline command/delegate/delegate.go:479]: Drop one of the 'LE' you can tell from the package what type of client it is

Comment (eoinmcafee00) [reply command/delegate/delegate.go:479]: and vice versa

Comment (tphoney) [reply command/delegate/delegate.go]: changed.

**Review by eoinmcafee00 — APPROVED** [review]

**Pull request merged**